course5/homework03/homework03.cc

```diff
@@ -1,6 +1,35 @@
 #include <cstdio>
+#include <string>
 
-// TODO your code goes here
+void print(int i)
+{
+    std::printf("%d", i);
+}
+
+void print(bool b)
+{
+    std::printf(b ? "true" : "false");
+}
+
+void print(double d)
+{
+    std::printf("%lg", d);
+}
+
+void print(double d, unsigned precision)
+{
+    std::printf("%.*lf", precision, d);
```

Comment:
I was under the impression that %f is for float, and %lf is for double. Apperently I was wrong, and older compilers even reject %lf and float is always converted to double when passed to printf. Since C99 and C++11 the standard says %f and %lf is the same thing.